private/buf/cmd/buf/buf.go

```diff
@@ -186,31 +197,29 @@ func NewRootCommand(name string) *appcmd.Command {
 						Deprecated: imageDeprecationMessage,
 						Hidden:     true,
 						SubCommands: []*appcmd.Command{
-							convert.NewCommand(
-								"convert",
-								builder,
-								imageDeprecationMessage,
-								true,
-							),
+							appcmd.NewDeletedCommand("convert", "Command deleted."),
 						},
 					},
-					push.NewCommand("push", builder, betaPushDeprecationMessage, true),
+					appcmd.NewDeletedCommand("push", betaPushDeprecationMessage),
 					{
 						Use:        "mod",
 						Short:      "Configure and update buf modules.",
 						Deprecated: betaModDeprecationMessage,
 						Hidden:     true,
 						SubCommands: []*appcmd.Command{
-							modinit.NewCommand("init", builder, betaModDeprecationMessage, true),
-							modupdate.NewCommand("update", builder, betaModDeprecationMessage, true),
-							export.NewCommand("export", builder, betaModExportDeprecationMessage, true),
-							modclearcache.NewCommand("clear-cache", builder, betaModDeprecationMessage, true, "cc"),
+							appcmd.NewDeletedCommand("init", betaModInitDeprecationMessage),
+							appcmd.NewDeletedCommand("update", betaModUpdateDeprecationMessage),
+							appcmd.NewDeletedCommand("export", betaModExportDeprecationMessage),
+							appcmd.NewDeletedCommand("clear-cache", betaModClearCacheDeprecationMessage, "cc"),
 						},
 					},
 					{
 						Use:   "registry",
 						Short: "Interact with the Buf Schema Registry.",
 						SubCommands: []*appcmd.Command{
+							// TODO(alex): Move these packages into registry/{login,logout}.
+							login.NewCommand("login", builder, "", false),
+							logout.NewCommand("logout", builder, "", false),
```

Comment:
The justification here is that we should aggressively protect the top-level command space (to prevent ourselves from being a `docker`). It's much easier to hoist a sub-command into the top-level than the reverse.

Here, we can actually move `login` and `logout` to `buf registry {login,logout}` rather than starting them in `buf beta registry {login,logout}`. Then, as we move out the rest of the `buf beta registry ...` commands, they will exist alongside `buf registry {login,logout}`.

`buf registry` should be enough of a top-level command to garner attention.

Reply:
As I've thought about this one a little bit more, it might actually make sense to leave `login` and `logout` at the top-level. Similar to our discussion around `buf config init` being a general configuration command that will (eventually) apply to `buf.gen.yaml`, `buf.work.yaml` and others, the same could be said about `buf login`.

The `buf login` command already authenticates users into _both_ the BSR and the Go Module Proxy. As we roll out more server-side components that require authentication, the list of things that `buf login` could be responsible for will grow.

These components might have their own authentication mechanisms, so it might make sense to keep each `login` command isolated in their own respective sub-commands (e.g. `buf registry login`, etc.), but I'm not sure how likely that is - it would end up being a confusing user experience to say the least (i.e. "THIS token is for this, and THAT token is for this other thing").

Anyway, this is just some additional food for thought. As my original comment suggests, it's a lot easier to hoist a sub-command into the top-level than the reverse, so we can start with `buf registry {login,logout}` and adapt later if need be.

Reply:
Discussed offline - starting with `buf registry {login,logout}` is best for now given all the context above.